### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by nilsdeppe**

### CODE REVIEW

File: src/DataStructures/DataBox.hpp [hunk 1]
```diff
@@ -232,52 +235,52 @@ class TaggedDeferredTuple
...
              (not tuples_detail::all<std::is_convertible<
-                  Tuple, item_type<Tags>>::value...>::value and
+                  Tuple, Deferred<item_type<Tags>>>::value...>::value and
               not tuples_detail::all<std::is_constructible<
-                  item_type<Tags>, Tuple>::value...>::value and
-              not tuples_detail::all<
-                  std::is_same<item_type<Tags>, Ts>::value...>::value);
+                  Deferred<item_type<Tags>>, Tuple>::value...>::value and
+              not tuples_detail::all<std::is_same<Deferred<item_type<Tags>>,
+                                                  Ts>::value...>::value);
```
Comment (kidder) [inline src/DataStructures/DataBox.hpp]: is_same_v

File: src/DataStructures/DataBox.hpp [hunk 2]
```diff
@@ -232,52 +235,52 @@ class TaggedDeferredTuple
...
```
Comment (kidder) [inline src/DataStructures/DataBox.hpp]: is_same_v

File: src/Utilities/TaggedTuple.hpp [hunk 3]
```diff
@@ -155,12 +155,13 @@ class TaggedTupleLeaf<Tag, false> {
...
-            typename std::enable_if<
-                !std::is_same<typename std::decay<T>::type,
-                              TaggedTupleLeaf>::value &&
-                std::is_constructible<value_type, T&&>::value>::type* = nullptr>
-  constexpr explicit TaggedTupleLeaf(T&& t) noexcept(
+  // clang-tidy: forwarding references can be bad
+  template <
+      class T,
+      typename std::enable_if<
+          !std::is_same<typename std::decay<T>::type, TaggedTupleLeaf>::value &&
```
Comment (kidder) [inline src/Utilities/TaggedTuple.hpp:162]: is_same_v

File: src/Utilities/TaggedTuple.hpp [hunk 4]
```diff
@@ -202,12 +203,12 @@ class TaggedTupleLeaf<Tag, true> : private Tag::type {
...
-  template <class T,
-            typename std::enable_if<
-                !std::is_same<typename std::decay<T>::type,
-                              TaggedTupleLeaf>::value &&
-                std::is_constructible<value_type, T&&>::value>::type* = nullptr>
-  constexpr explicit TaggedTupleLeaf(T&& t) noexcept(
+  template <
+      class T,
+      typename std::enable_if<
+          !std::is_same<typename std::decay<T>::type, TaggedTupleLeaf>::value &&
```
Comment (kidder) [inline src/Utilities/TaggedTuple.hpp:209]: is_same_v

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/DataStructures/DataBox.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/DataStructures/DataBox.hpp]: I'm keeping the TaggedTuple and TaggedDeferredTuple portions the same as their [standalone versions](https://github.com/nilsdeppe/tuples), which means C++11 compliant

**Comment by nilsdeppe** [comment]
The clang-tidy warning is something @osheamonn should look into

File: src/DataStructures/DataBox.hpp (hunk 1 above)
Comment (kidder) [reply src/DataStructures/DataBox.hpp]: okay, but this in in DataBox.hpp

**Review by kidder — DISMISSED** [review]

File: src/DataStructures/DataBox.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/DataStructures/DataBox.hpp]: Sure, that's where TaggedDeferredTuple lives, though. Do you want me to move that to a separate file?

Comment (kidder) [reply src/DataStructures/DataBox.hpp]: yes, if you want to keep certain things c++11 compliant, I think it makes sense to keep them separate

Comment (nilsdeppe) [reply src/DataStructures/DataBox.hpp]: I would rather make DataBox C++11 compatible, but regardless of that, TaggedDeferredTuple needs to be associated with DataBox, so the natural place to factor it out to would `src/DataStructures/TaggedDeferredTuple.hpp`. Is that okay?

**Review by kidder — APPROVED** [review]

**Pull request merged**